### PR TITLE
refactored the CamelKNodeProvider

### PR DESCRIPTION
**Pull request by lhein**
Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

### CODE REVIEW

File: src/CamelKNodeProvider.ts [hunk 1]
```diff
@@ -49,92 +49,85 @@ export class CamelKNodeProvider implements vscode.TreeDataProvider<TreeNode> {
 	}
 
 	// get the list of children for the node provider
-	public getChildren(element?: TreeNode): Thenable<TreeNode[]> {
+	public getChildren(element?: TreeNode): Promise<TreeNode[]> {
```
Comment (apupier) [inline src/CamelKNodeProvider.ts]: what is the advantage of changing from Thenable to promise?

Comment (lhein) [reply src/CamelKNodeProvider.ts]: It returns a Promise and I see no benefit in using Thenable here. So why should we obscure the real return type?

Comment (apupier) [reply src/CamelKNodeProvider.ts]: I don't know why Thenable was used initially here.
Thenable seems more generic than Promise.
I have no idea what is the best practice, asking to try to understand it.

Comment (lhein) [reply src/CamelKNodeProvider.ts]: best I could find atm is https://stackoverflow.com/questions/29435262/regarding-promises-a-specification-what-is-the-difference-between-the-terms-t

I think you mistake it for the "better return the interface" paradigm in Java which I think doesn't fit here.

Comment (lhein) [reply src/CamelKNodeProvider.ts]: But to be perfectly honest I am not able to give you a 100% answer here. I am myself too new to JS/TS.

Comment (apupier) [reply src/CamelKNodeProvider.ts]: still don't understand what is best to use but at least we are more "coherent" with promise as it is what we are using most of the time in our codebase

**Comment by bfitzpat** [comment]
Closed with merging of combined PR https://github.com/camel-tooling/vscode-camelk/pull/141

**Pull request closed without merging**